src/rules.c

```diff
@@ -4135,6 +4141,54 @@ solver_addrecommendsrules(Solver *solv)
   solv->recommendsrules_end = solv->nrules;
 }
 
+/* add rules to exclude solvables provided by lower
+ * precedence repositories */
+void solver_addstrictrepopriorules(struct s_Solver *solv, Map *addedmap)
+{
+  Pool *pool = solv->pool;
+	Solvable *s;
+	Id p, p2, pp2;
+	Map priomap;
+	int max_prio;
+
+	map_init_clone(&priomap, addedmap);
+	solv->strictrepopriorules = solv->nrules;
+
+	FOR_POOL_SOLVABLES(p)
+	{
+		if (!MAPTST(&priomap, p))
+		  	continue;
+
+		s = pool->solvables + p;
+		max_prio = s->repo->priority;
+		FOR_PROVIDES(p2, pp2, s->name)
+		{
+		Solvable *s2 = pool->solvables + p2;
+		if (s->name != s2->name)
+		{
+		MAPCLR(&priomap, p2);
```

Comment:
why mapclr here?

Reply:
yes, that's wrong. you mustn't clear the bits in the first loop, as the second loop also does a MAPTST.

Reply:
Oh wait, it only clears if the names are different. It's still wrong, though ;-)

Reply:
Sorry I think I got it, from `conda` packages names of providers are always the same so I take a wrong shortcut to clear them if names differ.

In the case where `s->name != s2->name` we don't clear `s2` but just continue to not prevent to add rules for `s2->name` later right?